core-rust/state-manager/src/store/db.rs

```diff
@@ -258,6 +265,18 @@ impl TransactionIndex<&LedgerPayloadHash> for StateManagerDatabase {
     }
 }
 
+impl TransactionIdentifierLoader for StateManagerDatabase {
+    fn get_top_transaction_identifiers(&self) -> CommittedTransactionIdentifiers {
+        match self {
+            StateManagerDatabase::InMemory(store) => store.get_top_transaction_identifiers(),
+            StateManagerDatabase::RocksDB(store) => store.get_top_transaction_identifiers(),
+            // This is a special case, where in tests, the Core API needs to construct a state
+            // manager with no database (stuck at pre-genesis).
```

Comment:
_Note: literally 1 test (`test_core_api_server_start_and_stop()`) needs this special-casing - can we avoid it?_

Reply:
This question came up before. Probably we should just ditch it (and the test as well).

Reply:
https://github.com/radixdlt/babylon-node/blob/b464c5594d1dc2dfd6dc1eb823d6e9f4a804d88e/core/src/test/java/com/radixdlt/api/CoreApiServerTest.java#L81

I guess we are at "some point" right now!
Would it also make `DatabaseConfig.NONE` deletable?

Reply:
:+1: for removing `CoreApiServerTest` and `DatabaseConfig.NONE`

Reply:
I think it's just this one test ATM.

Thanks for 🟢 , this will happen in its own PR.